### PR TITLE
Fix/bug fixing

### PR DESCRIPTION
**Pull request by juliswer**

### CODE REVIEW

File: src/containers/create-new/index.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { selectNftState } from '../../features/leda-nft/store/leda-nft.slice';
 import useMetamask from '../../features/auth/hooks/useMetamask';
 import { ItemProperty } from '../../common/types/ipfs-types';
 
-const tagsErrorMessage = {
+const tagsErrorMessages = {
   CantMore: 'You can not enter more than 8 tags',
   AtLeast: 'Please enter at least 1 tag',
   LenghtNotAllowed: 'Too long tag. Try with a tag that contains less than 8 characters',
```
Comment (GimerStackIt) [inline src/containers/create-new/index.tsx:21]: I suggest shorter messages.

File: src/containers/create-new/index.tsx [hunk 2]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
...
     if (isPreviewBtn && selectedImage) {
       setPreviewData({ ...data, blob: selectedImage });
       setShowProductModal(true);
     }
-    if (!isPreviewBtn && selectedImage && tags.length > 0 && tags.length <= 8) {
+    if (
+      !isPreviewBtn &&
+      selectedImage &&
+      tags.length > 0 &&
+      tags.length <= 8 &&
```
Comment (GimerStackIt) [inline src/containers/create-new/index.tsx:134]: Should we hard code this value (8)?

File: src/containers/create-new/index.tsx [hunk 3]
```diff
@@ -139,6 +148,8 @@ const CreateNewArea = () => {
 
   const handleTagsChange = (tagProps: string[]) => {
     // prevent empty tags
+    /* const itemBeingAdded = tagProps[tagProps.length - 1];
+    if (itemBeingAdded.length >= 8) setTagErrMessage(tagsErrorMessages.LenghtNotAllowed); */
```
Comment (GimerStackIt) [inline src/containers/create-new/index.tsx:152]: Remove comments

**Review by GimerStackIt — COMMENTED** [review]
I left a few comments.

File: src/assets/css/bootstrap.min.css [hunk 4]
```diff
@@ -6023,6 +6023,7 @@ fieldset:disabled .btn {
 }
 .modal-body {
   position: relative;
+  margin: auto;
```
Comment (mmiranda-stackit) [inline src/assets/css/bootstrap.min.css:6026]: I'd highly suggest to not modify libraries source code (I mean, even though we have it locally, imagine that in a future, someone that is not aware of the custom css in this file decides to upgrade the version to something newer. The custom css will be gone). What can I do? Create a custom css file and override the bootstrap styles. That way we'll make sure our own rules wi'll stay there no matter what changes we do to the main library file.

File: src/components/item/index.tsx [hunk 5]
```diff
@@ -55,6 +55,7 @@ const Product = ({
   imageQuality = 85,
   isCreator = false,
   tagsCreatePage,
+
```
Comment (mmiranda-stackit) [inline src/components/item/index.tsx:58]: useless space

File: src/components/item/index.tsx [hunk 6]
```diff
@@ -88,7 +89,11 @@ const Product = ({
               />
             </Anchor>
           ) : (
-            <img src={imageString} alt="Leda MarketPlace." />
+            <img
+              src={imageString}
+              alt="Leda MarketPlace."
+              style={{ maxWidth: '400px', maxHeight: '800px' }}
```
Comment (mmiranda-stackit) [inline src/components/item/index.tsx:95]: I"m ok with inline style. However, if this file already has a css file, use it. Let's keep the components as clean as possible

File: src/containers/activity/activity-area.tsx [hunk 7]
```diff
@@ -24,16 +25,17 @@ export const ActivityArea = () => {
     );
   }
 
-  if (!history) {
-    return <>Loading</>;
+  if (isLoadingHistory) {
+    return (
+      <div className="text-center my-4">
+        <ClipLoader className="spinner" color="#35b049" />
```
Comment (mmiranda-stackit) [inline src/containers/activity/activity-area.tsx:31]: Is this color good in both dark and light mode?

File: src/containers/author-profile/index.tsx [hunk 8]
```diff
@@ -24,6 +25,10 @@ const AuthorProfileArea = ({ className, address }: Props) => {
   const onSaleItems = useAppSelector((state) => selectOnSaleItems(state, address));
   const ownedItems = useAppSelector((state) => selectOwnedItems(state, address));
 
+  const likedItemsToShow = likedItems.filter(
```
Comment (mmiranda-stackit) [inline src/containers/author-profile/index.tsx:28]: use a memo here so you avoid the calculation in every re-render

File: src/containers/hero/hero.tsx [hunk 9]
```diff
@@ -1,15 +1,39 @@
-import Button from '@ui/button';
+// * Import React Hooks
```
Comment (mmiranda-stackit) [inline src/containers/hero/hero.tsx:1]: no need of this comments. They make the reading harder and they're self explanatory

File: src/containers/hero/hero.tsx [hunk 10]
```diff
@@ -1,15 +1,39 @@
...
   homeSection?: HomeSection;
 };
 
+const LoadingSpinner = () => (
+  <div className="text-center">
+    <ClipLoader className="spinner" color="#35b049" />
+  </div>
+);
+
+// ? Render Hero Component
```
Comment (mmiranda-stackit) [inline src/containers/hero/hero.tsx:25]: same with this comment and rest of them. I'd suggest to not include anything like this.

File: src/containers/hero/hero.tsx [hunk 11]
```diff
@@ -65,21 +89,25 @@ const Hero = ({ homeSection }: Props) => {
...
-                    tokenId={item.tokenId}
-                    price={Number(item.price)}
-                    tags={item.tags}
-                    status={item.status}
-                    likeCount={item.likes}
-                    imageString={item.image.url}
-                  />
-                </div>
-              ))}
+              {loadingNewest ? (
```
Comment (mmiranda-stackit) [inline src/containers/hero/hero.tsx:92]: Can't we use the Spinner component as a container ?

File: src/features/leda-nft/services/item.service.ts [hunk 12]
```diff
@@ -18,6 +18,11 @@ export default class ItemService extends HttpService implements IItemService {
     return data;
   }
 
+  async getNewest(qty: number): Promise<Item[]> {
+    const { data } = await this.instance.get(`${this.endpoint}/newest/${qty}`);
```
Comment (mmiranda-stackit) [inline src/features/leda-nft/services/item.service.ts:22]: I would make the endpoint to look like this: api/items/newest?qty=2. All the filtering criteria should be in the query string. Note: Make sure to have a max amount allowed in the backend, so the endpoint is not abusively used.

File: src/containers/create-new/index.tsx [hunk 13]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
...
       setPreviewData({ ...data, blob: selectedImage });
       setShowProductModal(true);
     }
-    if (!isPreviewBtn && selectedImage && tags.length > 0 && tags.length <= 8) {
+    if (
+      !isPreviewBtn &&
+      selectedImage &&
+      tags.length > 0 &&
+      tags.length <= 8 &&
+      longTags.length === 0
```
Comment (AlefrankM) [inline src/containers/create-new/index.tsx:135]: You could avoid this with !!longTags.length

File: src/containers/create-new/index.tsx [hunk 14]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
...
+    if (tags.length === 0) setTagErrMessage(tagsErrorMessages.AtLeast);
     if (isPreviewBtn && selectedImage) {
       setPreviewData({ ...data, blob: selectedImage });
       setShowProductModal(true);
     }
-    if (!isPreviewBtn && selectedImage && tags.length > 0 && tags.length <= 8) {
+    if (
+      !isPreviewBtn &&
+      selectedImage &&
+      tags.length > 0 &&
```
Comment (AlefrankM) [inline src/containers/create-new/index.tsx:133]: You could get rid of this > 0

File: src/containers/create-new/index.tsx [hunk 15]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
     const submitBtn = target.localName === 'span' ? target.parentElement : target;
     const isPreviewBtn = submitBtn.dataset?.btn;
     setHasImageError(!selectedImage);
-    if (tags.length > 8) setTagErrMessage(tagsErrorMessage.CantMore);
-    if (tags.length === 0) setTagErrMessage(tagsErrorMessage.AtLeast);
+    const longTags = tags.filter((tag) => tag.length >= 8);
+
+    if (longTags.length > 0) setTagErrMessage(tagsErrorMessages.LenghtNotAllowed);
```
Comment (AlefrankM) [inline src/containers/create-new/index.tsx:123]: You can use longTags.length without > 0

File: src/containers/create-new/index.tsx [hunk 16]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
...
+    if (tags.length > 8) setTagErrMessage(tagsErrorMessages.CantMore);
+    if (tags.length === 0) setTagErrMessage(tagsErrorMessages.AtLeast);
```
Comment (AlefrankM) [inline src/containers/create-new/index.tsx:125]: You can use !!longTags.length without === 0

File: src/containers/create-new/index.tsx [hunk 17]
```diff
@@ -118,13 +118,22 @@ const CreateNewArea = () => {
     const submitBtn = target.localName === 'span' ? target.parentElement : target;
     const isPreviewBtn = submitBtn.dataset?.btn;
     setHasImageError(!selectedImage);
-    if (tags.length > 8) setTagErrMessage(tagsErrorMessage.CantMore);
-    if (tags.length === 0) setTagErrMessage(tagsErrorMessage.AtLeast);
+    const longTags = tags.filter((tag) => tag.length >= 8);
+
+    if (longTags.length > 0) setTagErrMessage(tagsErrorMessages.LenghtNotAllowed);
+    if (tags.length > 8) setTagErrMessage(tagsErrorMessages.CantMore);
```
Comment (AlefrankM) [inline src/containers/create-new/index.tsx:124]: You can use a constant

File: src/containers/hero/hero.tsx [hunk 18]
```diff
@@ -1,15 +1,32 @@
...
+  </div>
+);
+
 const Hero = ({ homeSection }: Props) => {
-  const newItems = useAppSelector(selectNewest);
+  const dispatch = useAppDispatch();
+  const { newestItems, loadingNewest } = useAppSelector(selectMarketplaceState);
+
+  useEffect(() => {
+    dispatch(getNewest(2));
```
Comment (AlefrankM) [inline src/containers/hero/hero.tsx:27]: Can you avoid this MagicNumber?

File: src/containers/hero/hero.tsx [hunk 19]
```diff
@@ -64,23 +81,25 @@ const Hero = ({ homeSection }: Props) => {
...
-                  />
-                </div>
-              ))}
-            </div>
+            <SpinnerContainer isLoading={loadingNewest}>
+              <div className="row g-5">
+                {newestItems.map((item: ItemType) => (
+                  <div className="col-md-6" key={item.itemId}>
+                    <Item
+                      title={item.name}
```
Comment (AlefrankM) [inline src/containers/hero/hero.tsx:89]: It's posible avoid this prop drilling with the global state?

**Review by AlefrankM — COMMENTED** [review]
Good

**Review by mmiranda-stackit — APPROVED** [review]

**Pull request merged**